src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -229,7 +241,19 @@ public RestoreCommand()
 
                 // Convert package sources to repositories
                 var sourceProvider = GetSourceRepositoryProvider();
-                var repositories = GetPackageSources(Settings).Select(source => sourceProvider.CreateRepository(source));
+
+                var packageSources = GetPackageSources(Settings);
```

Comment:
Can probably move getting the packages sources outside the loop, and then print it once